src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -134,11 +134,15 @@ internal static MethodInfo GetCompileMethod(Type lambdaExpressionType)
         /// Produces a delegate that represents the lambda expression.
         /// </summary>
         /// <returns>A delegate containing the compiled version of the lambda.</returns>
+        [RequiresDynamicCode(Expression.NewArrayRequiresDynamicCode)]
```

Comment:
This message is a bit weird (IMO). Anytime someone tries to compile an expression, they will get the warning 

> "Creating arrays at runtime requires dynamic code generation. This warning can be suppressed if there are no new array nodes in the expression tree."

But won't they already get the warning from the call that actually creates new array nodes in the expression tree? Therefore, we don't need to also warn them at Compile().

Reply:
Hmm, hadn't thought of that. The problem is that something like `Expression<Func<int[]>> e = () => new[] { 1 };` won't produce an analyzer warning since we don't see the call site, but maybe that can be OK for now and we can treat it as an analyzer improvement to fix later.

Reply:
In that case, wouldn't it be fine? There's code there for a `new int[]`.

Reply:
Well, maybe. Replace `int` with your favorite custom struct. That likely wouldn't work.